pkg/influx/api_test.go

```diff
@@ -0,0 +1,165 @@
+package influx
+
+import (
+	"bytes"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+	"github.com/go-kit/log"
+	"github.com/grafana/influx2cortex/pkg/errorx"
+	"github.com/grafana/influx2cortex/pkg/remotewrite/remotewritemock"
+	"github.com/stretchr/testify/assert"
+	mock "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleSeriesPush(t *testing.T) {
+	tests := []struct {
+		name            string
+		url             string
+		data            string
+		expectedCode    int
+		remoteWriteMock func() *remotewritemock.Client
+		recorderMock    func() *MockRecorder
+	}{
+		{
+			name:         "POST",
+			url:          "/write",
+			data:         "measurement,t1=v1 f1=2 1465839830100400200",
+			expectedCode: http.StatusNoContent,
+			remoteWriteMock: func() *remotewritemock.Client {
+				remoteWriteMock := &remotewritemock.Client{}
+				remoteWriteMock.On("Write", mock.Anything, &cortexpb.WriteRequest{
+					Timeseries: []cortexpb.PreallocTimeseries{
+						{
+							TimeSeries: &cortexpb.TimeSeries{
+								Labels: []cortexpb.LabelAdapter{
+									{Name: "__name__", Value: "measurement_f1"},
+									{Name: "t1", Value: "v1"},
+								},
+								Samples: []cortexpb.Sample{
+									{Value: 2, TimestampMs: 1465839830100},
+								},
+							},
+						},
+					},
+				}).Return(nil)
+				return remoteWriteMock
+			},
+			recorderMock: func() *MockRecorder {
+				recorderMock := &MockRecorder{}
+				recorderMock.On("measureMetricsParsed", 1).Return(nil)
+				recorderMock.On("measureMetricsWritten", 1).Return(nil)
+				recorderMock.On("measureConversionDuration", mock.Anything).Return(nil)
```

Comment:
can you confirm that the duration is at least not zero?

Reply:
Used mock.MatchedBy to make sure the number is a positive number when expected